### DISCUSSION TITLE
3.3.1  BUILD FAILED (macOS 14.4.1 using ruby-build 20240423)

### DISCUSSION THREAD

**Discussion by stadia** [rbenv/ruby-build#2371] [Build failures] [answered]
macOS Sonoma 14.4.1

while build it 3.3.1, the errors happened in Rust YJIT section

```
building Rust YJIT (release mode)
dyld[30190]: Symbol not found: __ZN4llvm10PGOOptionsC1ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEES7_S7_S7_NS_18IntrusiveRefCntPtrINS_3vfs10FileSystemEEENS0_9PGOActionENS0_11CSPGOActionEbb
  Referenced from: <EEB49284-2EAD-3371-9401-139E8557678B> /opt/homebrew/Cellar/rust/1.77.2_1/lib/librustc_driver-3dc17bded25d4e59.dylib
  Expected in:     <B0C4597F-D1E6-36D5-83FD-81D0ACE34F6C> /opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib
make: *** [yjit/target/release/libyjit.a] Abort trap: 6
make: *** Waiting for unfinished jobs....
external command failed with status 2
```

**Comment by ryansdwilson** [comment]
I am also getting the same error.

[ruby-build.20240425093458.4257.log](https://github.com/rbenv/ruby-build/files/15113125/ruby-build.20240425093458.4257.log)

**Comment by Cueball** [comment]
Similar error, but I'm still on x64 (paths are different, and it's not a architecture-specific error):
```
building Rust YJIT (release mode)
dyld[75212]: Symbol not found: __ZN4llvm10PGOOptionsC1ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEES7_S7_S7_NS_18IntrusiveRefCntPtrINS_3vfs10FileSystemEEENS0_9PGOActionENS0_11CSPGOActionEbb
  Referenced from: <728B363C-C125-3F29-BF64-36E1FD9A5755> /usr/local/Cellar/rust/1.77.2_1/lib/librustc_driver-439314449a904cb0.dylib
  Expected in:     <83624E46-48B6-3467-AE49-9DBD1E457F64> /usr/local/Cellar/llvm/18.1.4/lib/libLLVM.dylib
make: *** [yjit/target/release/libyjit.a] Abort trap: 6
make: *** Waiting for unfinished jobs....
external command failed with status 2
```

**Comment by hsbt** [comment]
Does anyone try install without `llvm` of homebrew?

**Reply by stadia** [reply]
ruby-build needs the rust then it need the llvm@17

**Reply by mislav** [reply]
I was just able to reproduce the problem. First, I had a previous installation of rust (and its Homebrew dependency `llvm@17`) installed, and using that combination Ruby 3.3.1 compilation succeeded.

But, then I did `brew reinstall rust` which resulted in llvm 18.1.4 getting installed as a dependency. Then, building Ruby 3.3.1 with `--enable-yjit` configuration flag produces the following failure:
```
building Rust YJIT (release mode)
dyld[82704]: Symbol not found: __ZN4llvm10PGOOptionsC1ENSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEES7_S7_S7_NS_18IntrusiveRefCntPtrINS_3vfs10FileSystemEEENS0_9PGOActionENS0_11CSPGOActionEbb
  Referenced from: <EEB49284-2EAD-3371-9401-139E8557678B> /opt/homebrew/Cellar/rust/1.77.2_1/lib/librustc_driver-3dc17bded25d4e59.dylib
  Expected in:     <B0C4597F-D1E6-36D5-83FD-81D0ACE34F6C> /opt/homebrew/Cellar/llvm/18.1.4/lib/libLLVM.dylib
make: *** [yjit/target/release/libyjit.a] Abort trap: 6
```

Explicitly running `brew install llvm@17` makes the Ruby compilation issue go away even though Homebrew `llvm` is still at version 18.

Tracking as https://github.com/rbenv/ruby-build/issues/2372

**Comment by stadia** [comment, marked as answer]
Got it. If we install the latest rust via homebrew, then rust has the llvm@17 as a dependency.
But we have installed the llvm instead of the llvm@17. So we just install the llvm@17

**Reply by ryansdwilson** [reply]
brew reinstall rust rustup worked for me
